components/inputs/input-textarea.js

```diff
@@ -0,0 +1,261 @@
+					?disabled="${disabled}"
+					id="${this._textareaId}"
+					@input="${this._handleInput}"
+					@invalid="${this._handleInvalid}"
+					maxlength="${ifDefined(this.maxlength)}"
+					minlength="${ifDefined(this.minlength)}"
+					placeholder="${ifDefined(this.placeholder)}"
+					?required="${this.required}">${this.value}</textarea>
+				${this.validationError ? html`<d2l-tooltip for=${this._textareaId} state="error" align="start">${this.validationError}</d2l-tooltip>` : null}
+			</div>
+			${offscreenContainer}
+		`;
+
+		if (this.label && !this.labelHidden) {
+			return html`
+				<label class="d2l-input-label d2l-skeletize" for="${this._textareaId}">${this.label}</label>
+				${textarea}`;
+		}
+
+		return textarea;
+
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		changedProperties.forEach((oldVal, prop) => {
+			if (prop === 'value') {
+				this.setValidity({ tooShort: this.minlength && this.value.length > 0 && this.value.length < this.minlength });
+				this.requestValidate(false);
+				this.setFormValue(this.value);
+				this._prevValue = (oldVal === undefined) ? '' : oldVal;
+			} else if (prop === 'validationError') {
+				if (oldVal && this.validationError) {
+					const tooltip = this.shadowRoot.querySelector('d2l-tooltip');
+					tooltip.updatePosition();
+				}
+			}
+		});
+	}
+
+	async focus() {
+		const elem = this.shadowRoot.querySelector('textarea');
+		if (elem) {
+			elem.focus();
+		} else {
+			await this.updateComplete;
+			this.focus();
+		}
+	}
+
+	get validationMessage() {
+		if (this.validity.tooShort) {
+			return this.localize('components.form-element.input.text.tooShort', { label: this.label, minlength: formatNumber(this.minlength) });
+		}
+		return super.validationMessage;
+	}
+
+	get validity() {
+		const elem = this.shadowRoot.querySelector('textarea');
+		if (!elem.validity.valid) {
+			return elem.validity;
+		}
+		return super.validity;
+	}
+
+	_getAriaLabel() {
+		if (this.label && this.labelHidden) {
+			return this.label;
+		}
+		if (this.hasAttribute('aria-label')) {
+			return this.getAttribute('aria-label');
+		}
+		return undefined;
+	}
+
+	async _handleBlur(e) {
+		this.requestValidate(true);
+
+		/**
+		 * This is needed only for legacy Edge since the _handleChange function is NOT
+		 * triggered, therefore we have to detect the blur and handle it ourselves.
+		 */
+		const browserType = window.navigator.userAgent;
+		if (this._prevValue !== e.target.value && (browserType.indexOf('Trident') > -1 || browserType.indexOf('Edge') > -1)) {
+			this._handleChange();
+		}
+	}
+
+	_handleChange() {
+		// change events aren't composed, so we need to re-dispatch
+		this.dispatchEvent(new CustomEvent(
+			'change',
+			{ bubbles: true, composed: false }
+		));
+	}
+
+	_handleInput(e) {
+		this.value = e.target.value;
+		return true;
+	}
+
+	_handleInvalid(e) {
+		e.preventDefault();
+	}
+
+}
+
+customElements.define('d2l-input-textarea', InputTextArea);
```

Comment:
I chose this name to be consistent with the format of the rest of the components in the inputs folder.  `d2l-inout-*`.  If anyone has objection I'd be open to changing this to `d2l-textarea`, which would align more closely to the underlying element but be inconsistent with our other components. (ex. `d2l-input-fieldset` is one such similar case).

Reply:
I'm OK either way here. I do like that all the inputs are grouped together like this, but this is the first one I believe that isn't natively `<input type="something">`.